### ISSUE TITLE
Issu with "power.prune() function"

### ISSUE THREAD

**Issue by RoaaNaa** [MRCIEU/TwoSampleMR#80]
Hello;
when i use this function in" Two sample MR package" i got this error  
### **_**### 
Error in order(dat$ncase.outcome, decreasing = T) : 
  argument 1 is not a vector
**_**
I tried to convert the data frame to vector but still there is Error:
"Error: $ operator is invalid for atomic vectors" 
Could you help me to solve the problem ?
thanks.

**Comment by mightyphil2000** [comment]
Hi Roaa


It looks there could be a bug in the code. I will take a look and get back to you but it could be a week.


In the meantime I would recommend pruning your duplicate outcomes on sample size or number of cases. This can be achieved by ordering your dataset by sample size using the order command with the decreasing argument set to TRUE. Then use the !duplicated command. e.g.


(assuming only one exposure in Data but multiple duplicate outcome traits with same name):

Data<-Data[order(Data$ncase,decreasing=T),]

Data<-Data[!duplicated(Data$outcome),]


bw

P

________________________________

**Comment by RoaaNaa** [comment]
Thank you alot

**Comment by mightyphil2000** [comment]
Hi Roaa


The power.prune function has now been updated. Let me know if it works for you now. You would first have to update the R package.


bw

Philip

________________________________

**Comment by ulysses1881826** [comment]
It seems this bug still exists. Here is my code which can be reproducible everywhere:
##
Lan.exposure_data <- subset(gwas_catalog, PubmedID == "30804566")
Lan.exposure_data$samplesize=78990
Lan.exposure_data <- format_data(Lan.exposure_data)
Lan.exposure_data <- clump_data(Lan.exposure_data)
Lan.outcome_data <- extract_outcome_data(snps = Lan.exposure_data$SNP,outcomes = 297)
Lan.dat <- harmonise_data(exposure_dat = Lan.exposure_data, outcome_dat = Lan.outcome_data)
Lan.dat <- power_prune(Lan.dat)
##

And the error is:
##
Lan.dat <- power_prune(Lan.dat)
[1] "finding summary set for --Insomnia symptoms (never/rarely vs. usually) Alzheimer's disease-- with largest sample size"
Error in order(ncase, decreasing = T) : argument 1 is not a vector
##

I look into the souce code of function "power_prune" and I cannot understand the code here:
##
if(any(is.na(ncase))){
				ncase<-dat1$samplesize.outcome
				if(dist.outcome=="binary") warning(paste("dist.outcome set to binary but case sample size is missing. Will use total sample size instead but power pruning may be less accurate"))
			}
##
the condition "any(is.na(ncase))" here returns false if ncase is NULL, is it the intention of the author to do so?

Thanks in advance

**Comment by mightyphil2000** [comment]
Hi Ulysses

Thanks for letting us know. I'm looking into this and will get back to you hopefully tomorrow.

bw
Philip
________________________________

**Comment by ulysses1881826** [comment]
Hello, I am here to ask about the process of the possible bug. Have you ever deal with it? Thank you!

**Comment by mightyphil2000** [comment]
Yes i think it's fixed. can you redownload the package and try again?
________________________________

**Comment by ulysses1881826** [comment]
Thank you for your kind reply. However, when I did as you told here, I still met the same problem. Reproducible code here:

#########################
library(TwoSampleMR)
library(MRInstruments)
data(gwas_catalog)
Kar.exposure_data <- subset(gwas_catalog, PubmedID == "30643258") #input is defined
# Kar.exposure_data= read.csv('/home/ubuntu/mendelianProject/myMendelian/Kar.exposure_data.withSamplesizeNumberOfInstrumentsFstat_big10.csv',row.names = 'X')
Kar.exposure_data <- format_data(Kar.exposure_data)
Kar.exposure_data <- clump_data(Kar.exposure_data)
Kar.Lam.outcome_data <- extract_outcome_data(snps = Kar.exposure_data$SNP,outcomes = "ieu-a-297")
Kar.Lam.dat <- harmonise_data(exposure_dat = Kar.exposure_data, outcome_dat = Kar.Lam.outcome_data)
power_prune(Kar.Lam.dat)
#################
error reported as following:
###########################
[1] "finding summary set for --Risk-taking tendency (4-domain principal component model) (unit increase) Alzheimer's disease-- with largest sample size"
Error in order(nexp, decreasing = T) : argument 1 is not a vector
In addition: Warning message:
In power_prune(Kar.Lam.dat) :
  dist.outcome set to binary but case sample size is missing. Will use total sample size instead but power pruning may be less accurate
###################################
Thank you!
